### ISSUE TITLE
StatefulSets should not have kubectl annotation

### ISSUE THREAD

**Issue by mxey** [prometheus-operator/prometheus-operator#1725]
The StatefulSets created by the operator inherit all annotations from the Prometheus and Alertmanager resources. That includes the `kubectl.kubernetes.io/last-applied-configuration` annotation. When using `kubectl apply --prune`, it will delete the StatefulSets, because it assumes it created them. 

**What did you do?**

```
minikube start --kubernetes-version v1.11.1
kubectl apply -f https://raw.githubusercontent.com/coreos/prometheus-operator/v0.22.2/bundle.yaml
kubectl rollout status deployment prometheus-operator
kubectl apply -f https://raw.githubusercontent.com/coreos/prometheus-operator/v0.22.2/example/user-guides/getting-started/prometheus.yaml
kubectl get statefulset  prometheus-prometheus -o jsonpath='{.metadata.annotations}'
```

**What did you expect to see?**

```
map[prometheus-operator-input-checksum:d1d959b1edaabd9c22b7f182784f2d3b]
```

**What did you see instead? Under which circumstances?**

```
map[kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"monitoring.coreos.com/v1","kind":"Prometheus","metadata":{"annotations":{},"name":"prometheus","namespace":"default"},"spec":{"resources":{"requests":{"memory":"400Mi"}},"serviceAccountName":"prometheus","serviceMonitorSelector":{"matchLabels":{"team":"frontend"}}}}
 prometheus-operator-input-checksum:d1d959b1edaabd9c22b7f182784f2d3b]
```

**Environment**

* Prometheus Operator version:

	v0.22.2

* Kubernetes version information:

	Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.1", GitCommit:"b1b29978270dc22fecc592ac55d903350454310a", GitTreeState:"clean", BuildDate:"2018-07-18T11:37:06Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"darwin/amd64"}
        Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.1", GitCommit:"b1b29978270dc22fecc592ac55d903350454310a", GitTreeState:"clean", BuildDate:"2018-07-17T18:43:26Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster kind: 

	minikube start --kubernetes-version v1.11.1

* Manifests:

https://raw.githubusercontent.com/coreos/prometheus-operator/v0.22.2/bundle.yaml
https://raw.githubusercontent.com/coreos/prometheus-operator/v0.22.2/example/user-guides/getting-started/prometheus.yaml

* Prometheus Operator Logs:

```
ts=2018-08-01T11:17:29.729373522Z caller=main.go:116 msg="Starting Prometheus Operator version '0.22.2'."
level=info ts=2018-08-01T11:17:29.746105562Z caller=operator.go:174 component=alertmanageroperator msg="connection established" cluster-version=v1.11.1
level=info ts=2018-08-01T11:17:29.819911294Z caller=operator.go:294 component=prometheusoperator msg="connection established" cluster-version=v1.11.1
level=info ts=2018-08-01T11:17:33.965874043Z caller=operator.go:557 component=alertmanageroperator msg="CRD created" crd=Alertmanager
level=info ts=2018-08-01T11:17:34.073753863Z caller=operator.go:1138 component=prometheusoperator msg="CRD created" crd=Prometheus
level=info ts=2018-08-01T11:17:36.98717586Z caller=operator.go:190 component=alertmanageroperator msg="CRD API endpoints ready"
level=info ts=2018-08-01T11:17:38.090788672Z caller=operator.go:1138 component=prometheusoperator msg="CRD created" crd=ServiceMonitor
level=info ts=2018-08-01T11:17:38.137639181Z caller=operator.go:1138 component=prometheusoperator msg="CRD created" crd=PrometheusRule
level=info ts=2018-08-01T11:17:47.180678429Z caller=operator.go:310 component=prometheusoperator msg="CRD API endpoints ready"
level=info ts=2018-08-01T11:17:47.221370353Z caller=operator.go:737 component=prometheusoperator msg="sync prometheus" key=default/prometheus
level=info ts=2018-08-01T11:17:47.343906965Z caller=operator.go:737 component=prometheusoperator msg="sync prometheus" key=default/prometheus
level=info ts=2018-08-01T11:19:02.00281072Z caller=operator.go:737 component=prometheusoperator msg="sync prometheus" key=default/prometheus
```

**Comment by mxinden** [comment]
Interesting case that we haven't encountered yet. Thanks for debugging.

Sounds to me like we should just filter out the annotation `kubectl.kubernetes.io/last-applied-configuration` when creating the statefulset. What do you think @mxey?

**Comment by mxey** [comment]
@mxinden Yes. Maybe filter out all annotations starting with `kubectl.kubernetes.io/`.

**Comment by mxinden** [comment]
@mxey That sounds reasonable to me.

Would you like to give this a try and propose a patch? Shouldn't be a big change here:

- https://github.com/coreos/prometheus-operator/blob/master/pkg/prometheus/statefulset.go#L148
- https://github.com/coreos/prometheus-operator/blob/master/pkg/alertmanager/statefulset.go#L252

I am more than happy to help!

**Comment by paulfantom** [comment]
Possibly related to https://github.com/prometheus-operator/prometheus-operator/issues/2018

**Comment by simonpasquier** [comment]
Closed by #2944
